spring-integration-gcp/src/main/java/org/springframework/integration/gcp/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -70,8 +78,9 @@ private void receiveMessage(PubsubMessage message, AckReplyConsumer consumer) {
 		}
 
 		try {
+			String payload = message.getData().toStringUtf8();
```

Comment:
Isn't this related to the `Charset` in the `publish()` above?
What will happen here if I didn't use `UTF-8` there though?

Reply:
That's a great point, I meant to get back to this.
The only way of not doing this that I can think of is to add a header to `PubsubMessage` saying which encoding the payload is in, assuming it is a `String`.
We might even be able to add another `PubsubMessage` header with the class of the payload and that might allow us to do the Object -> JSON -> Object conversion, to support Object payloads.

Without any context about the payload, we have to make an assumption... WDYT?

Reply:
Well, all works for me, but we might need to follow best practice here with Google Pub/Sub.
Right, now let's stick with default `UTF-8` and reconsider content-type negotiation in the separate issue. 
Does it make sense ?